bindings/matrix-sdk-ffi/src/notification.rs

```diff
@@ -3,11 +3,22 @@ use std::sync::Arc;
 use matrix_sdk_ui::notification_client::{
     NotificationClient as MatrixNotificationClient,
     NotificationClientBuilder as MatrixNotificationClientBuilder,
+    NotificationItem as MatrixNotificationItem,
 };
 use ruma::{EventId, RoomId};
 
 use crate::{error::ClientError, event::TimelineEvent, helpers::unwrap_or_clone_arc, RUNTIME};
 
+#[derive(uniffi::Enum)]
+pub enum NotificationEvent {
+    Timeline {
+        event: Arc<TimelineEvent>,
```

Comment:
I would go as far as saying that instead of returning `TimelineEvent` directly we could remove this type altogether, and just have the Timeline directly wrap the `TimelineEventType`

Reply:
oh no I don't mean changing the enum, I just meant that TimelineEvent is probably now not useful anymore so Timeline could be like this:
```
Timeline {
  event: Arc<TimelineEventType>
}
```

Since the only reason to have TimelineEvent was to wrap the TimelineEventType and expose in a generic way the `event_id `and the `sender_id` which now we have already either directly in app or in the `SenderInfo`

Reply:
Ah, that sounds about right; there's just the timestamp that might be useful, at least it's been used on Android as @jmartinesp told me earlier. It could be optional in one of the metadata structures, or filled somewhere else.

Reply:
The only values we're taking from `TimelineEvent` in Android are `senderId` and `eventId`, but we already have those  in the push notification so we can just pass them to our mapper class instead.

Reply:
Cool; I'm happy to change this in a followup PR, since this one is already bulky. Added that to #2330